core/ipfs/utils.go

```diff
@@ -66,4 +67,19 @@ func GetFileHash(r io.Reader) (string, error) {
 
 }
 
+func DownloadIpfsItem(ctx context.Context, gatewayUrl string, cid cid.Cid) (io.ReadCloser, error) {
+	url := fmt.Sprintf("%s/ipfs/%s", gatewayUrl, cid.String())
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
 
+	client := http.Client{}
+
+	resp, err := client.Do(req.WithContext(ctx))
```

Comment:
@perfectmak 

What do you think about a timeout?

Reply:
Yeah, I initially added a 5 mins time-out, but realized that sometimes people might shared large files and it might take a while to download completely, so just resorted to make the request cancelable via the context. So if the request is cancelled, ideally the download should cancel as well.